src/test/java/org/cactoos/iterator/ImmutableTest.java

```diff
@@ -45,4 +46,22 @@ public void doesNotAllowRemovingOfElements() {
         immutable.next();
         immutable.remove();
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void decoratesNext() {
```

Comment:
@krzyk this test doesn't test the functionality of `next` method, please refactor it.

Reply:
@krzyk But your test does not prove that the method is decorated correctly. 
If I change `next `method like:
```java
public T next() {
       this.iterator.next();
       return this.iterator.next();
    }
```
Test will pass but it shouldn't.